algorithms/C/linked-lists/merge-two-linkedlists.c

```diff
@@ -0,0 +1,130 @@
+//Merging two Linked-List
+
+#include<stdio.h>
+#include<stdlib.h>
+
+/*Structure definition:
+     data : holds the integer value of the node
+     next : holds the pointer to the next node
+*/
+struct node
+{
+    int data;
+    struct node *next;
+}*la,*lb;                         
+//We have two linked-lists: la (List A) and lb (List B)
+
+
+//Function to create List A of given size
+void lista(int size)
+{
+    int a,i;
+    struct node *temp,*newnode;
+
+    //Head node of List A
+    la=(struct node*)malloc(sizeof(struct node));
+
+    //Getting values to enter into List A
+    printf("\nEnter the value for A: ");
+    scanf("%d",&a);
+    la->data=a;
+    la->next=NULL;
+    temp=la;
+    for(i=1; i<size; i++)
+    {
+        newnode=(struct node*)malloc(sizeof(struct node));
+        printf("Enter the value for A: ");
+        scanf("%d",&a);
+        newnode->data=a;
+        newnode->next=NULL;
+        temp->next=newnode;
+        temp=newnode;
+    }
+    printf("\n");
+}
+
+//Function to create List B of given size
+void listb(int size)
+{
+    int b,i;
+    struct node *temp,*newnode;
+
+    //Head node of List B
+    lb=(struct node*)malloc(sizeof(struct node));
+    
+    //Getting values to enter into List B
+    printf("\nEnter the value for B: ");
+    scanf("%d",&b);
+    lb->data=b;

```

Comment:
**Possible null dereference [[CWE-690]](https://cwe.mitre.org/data/definitions/690.html):** 

`lb` will point to `NULL` in case malloc() fails to allocate the memory ( either due to insufficient memory or memory corruption ) 

this is the case thought the source.
Always perform check after allocating memory dynamically to make sure you have a valid pointer there, you would rather terminate the program or handle this this case on your own instead of leaving a possible undefined behaviour in there.

**Fix:**
perform a check before dereferencing the pointer.

Reply:
Yes sure. I will make changes in my code regarding this.